src/Inciter/AMR/refinement.h

```diff
@@ -0,0 +1,779 @@
+#ifndef AMR_refinement_h
+#define AMR_refinement_h
+
+#include <algorithm>
+
+#include "tet_store.h"
+#include "node_connectivity.h"
+
+// TODO: make this have a base class to support multiple generator schemes
+// using the policy design pattern
+
+namespace AMR {
+
+    class refinement_t {
+        private:
+
+            const size_t DEFAULT_REFINEMENT_LEVEL = 0; //TODO: Is this in the right place?
+            const size_t MIN_REFINEMENT_LEVEL = DEFAULT_REFINEMENT_LEVEL;
+
+            // TODO: Tidy up edge store references here
+            tet_store_t* tet_store;
+            node_connectivity_t* node_connectivity;
```

Comment:
Just a note: try not to use reference semantics in class state - will be more complex and/or problematic with migration.